ygen/directory.go

```diff
@@ -133,6 +137,124 @@ func GetOrderedDirectories(directory map[string]*Directory) ([]string, map[strin
 	return orderedDirNames, dirNameMap, nil
 }
 
+// GetOrderedPathDirectories returns an alphabetically-ordered slice of
+// Directory names and a map of Directories keyed by their paths, so that each
+// directory can be processed in path-alphabetical order. This helps produce
+// deterministic generated code, and minimize diffs when compared with expected
+// output (i.e., diffs don't appear simply due to reordering of the Directory
+// maps).
+func GetOrderedPathDirectories(directory map[string]*Directory) []string {
```

Comment:
What are our thoughts on using generics here? I guess we should not until we do not support Go <1.18?

```go
func sortedStringMap[T any](m map[string]T) []string {
	s := []string{}
	for k := range m {
		s = append(s, k)
	}
	sort.Strings(s)
	return s
}
```

Reply:
Yeah I'm planning to not touch generics until 1.20 (next year) since ygot currently supports the latest 3 versions of Go. Another consideration is the Go team is saying generics is not completely production ready.

Reply:
Great - that sounds good to me too :-)